Empire - The Last Battle/Assets/Unity/Scripts/MonoBehaviours/OverworldManager.cs

```diff
@@ -354,17 +366,18 @@ void _StormshapersPlayer_OnCardRemoved(CardData card, int index)
 				// check for total KO
 				if (_GameStateHolder._ActivePlayer.PlayerArmy.GetActiveUnits().Count == 0 && _GameStateHolder._InactivePlayer.CastleProgress != 4) {
 
-					// revive half the units;
 					List<Unit> units = _GameStateHolder._ActivePlayer.PlayerArmy.GetKOUnits();
-					for(int i = 0; i < units.Count / 2; i++) {
-						units[i].Heal();
-					}
-					// move commander to start tile;
-					_OverworldUI.Enable();
-					TileData startTile = _GameStateHolder._ActivePlayer.Type == PlayerType.Battlebeard ? _Board._BBStartTile : _Board._SSStartTile;
-					_OverworldUI.ForceMoveCommander(startTile);
+
+					// revive half the units;
+					ReviveUnits(_GameStateHolder._ActivePlayer, units, units.Count / 2);
+					//Reset position
+					RestCommanderPosition();
 					return;
 				}
+				else if (isGameOverForPlayer(_GameStateHolder._ActivePlayer))
```

Comment:
Other if statement gets called if active units == 0 and castle progress != 4.
This one gets called if active units == 0 and castle progress == 4.